packages/xo-web/src/xo-app/settings/acls/index.js

```diff
@@ -80,92 +80,63 @@ const ACL_ACTIONS = [
-    createGetObjectsOfType('VM'),
-    createGetObjectsOfType('VM-snapshot'),
-    (hosts, networks, pools, srs, vms, snapshots) => ({
-      ...keyBy(hosts, 'id'),
-      ...keyBy(networks, 'id'),
-      ...keyBy(pools, 'id'),
-      ...keyBy(snapshots, 'id'),
-      ...keyBy(srs, 'id'),
-      ...keyBy(vms, 'id'),
-    })
-  )
-  return { xoObjects: getHighLevelObjects }
-})
-class AclTable extends Component {
-  componentWillMount () {
-    let subjects = {}
-    const refresh = (newSubjects = undefined) => {
-      newSubjects && (subjects = newSubjects)
-      const { xoObjects } = this.props
-      const { acls, roles } = this.state
-      const resolvedAcls = filter(
-        map(acls, ({ id, subject, object, action }) => ({
-          id,
-          subject: subjects[subject] || subject,
-          object: xoObjects[object] || object,
-          action: roles[action] || action,
-        })),
-        ({ subject, object, action }) =>
-          subject && object && action && object.type !== 'VM-snapshot'
-      )
-      this.setState({
-        resolvedAcls,
-      })
-    }
-
-    const unsubscribeAcls = subscribeAcls(acls =>
-      this.setState({ acls }, refresh)
-    )
-    const unsubscribeRoles = subscribeRoles(roles =>
-      this.setState({ roles: keyBy(roles, 'id') }, refresh)
-    )
-    const unsubscribeGroups = subscribeGroups(groups => {
-      groups = keyBy(groups, 'id')
-      refresh({
-        ...pickBy(subjects, subject => subject.type === 'user'),
-        ...groups,
-      })
-    })
-    const unsubscribeUsers = subscribeUsers(users => {
-      users = keyBy(users, 'id')
-      refresh({
-        ...pickBy(subjects, subject => subject.type === 'group'),
-        ...users,
-      })
-    })
-
-    this.componentWillUnmount = () => {
-      unsubscribeAcls()
-      unsubscribeGroups()
-      unsubscribeRoles()
-      unsubscribeUsers()
-    }
-  }
-
-  render () {
-    const { resolvedAcls = [] } = this.state
-
-    return isEmpty(resolvedAcls) ? (
-      <p>
-        <em>{_('aclNoneFound')}</em>
-      </p>
-    ) : (
-      <SortedTable
-        actions={ACL_ACTIONS}
-        collection={resolvedAcls}
-        columns={ACL_COLUMNS}
-      />
-    )
-  }
-}
+const AclTable = [
+  connectStore({
+    hosts: createGetObjectsOfType('host'),
+    networks: createGetObjectsOfType('network'),
+    pools: createGetObjectsOfType('pool'),
+    srs: createGetObjectsOfType('SR'),
+    vms: createGetObjectsOfType('VM'),
+    snapshots: createGetObjectsOfType('VM-snapshot'),
+  }),
+  addSubscriptions({
+    acls: subscribeAcls,
+    roles: subscribeRoles,
+    groups: subscribeGroups,
+    users: subscribeUsers,
+  }),
+  provideState({
+    initialState: () => ({
+      acls: [],
+    }),
+    computed: {
+      acls: (
+        { groups, roles, users },
+        { acls, hosts, networks, pools, srs, vms, snapshots }
+      ) =>
+        filter(
+          map(acls, ({ id, subject, object, action }) => ({
```

Comment:
Better use a computed for this and i think is preferable to use `[].filter()` instead for less dependence.

Reply:
A computed for what?

And, indeed I do favor native functions to avoid dependencies as much as possible.

Reply:
for the `map` to avoid filter the map each time an object in the collection change

Reply:
My bad, `map` create a new array on each call.